src/widgets/useLoadingStateFromStore.js

```diff
@@ -0,0 +1,43 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { setWidgetLoaders, removeWidgetLoaders } from '../redux/cartoSlice';
+
+export default function useLoadingStateFromStore(widgetId, filterIsByViewport) {
+  const dispatch = useDispatch();
+  const widgetLoaders = useSelector((state) => state.carto.widgetLoaders);
+  const isLoadingStateSet = useRef(false);
+
+  const setLoading = useCallback(
+    (isLoading) => {
+      if (widgetId) {
+        dispatch(
+          setWidgetLoaders({
+            widgetId,
+            isLoading
+          })
+        );
+      }
+    },
+    [widgetId]
+  );
+
+  useEffect(() => {
+    if (!widgetId) {
+      throw new Error('Widget "id" property is required.');
+    }
+
+    setLoading(filterIsByViewport);
+
+    return () => {
+      removeWidgetLoaders();
```

Comment:
After reading this again, I guess we should remove just loading state for the current widget. If we want to keep the simmetry with the 'add' option, which allows 'all' and also one specific widgetId, we should probably allow passing a widgetId list to removeWidgetLoaders

Reply:
Let's modify our action to allow removing specific `widgetId`s, it's true that a single widget component could be un-mounted.